src/simtools/camera/single_photon_electron_spectrum.py

```diff
@@ -180,3 +207,113 @@ def _get_input_data(self, input_file, frequency_column):
         with tempfile.NamedTemporaryFile(delete=False, mode="w", encoding="utf-8") as tmpfile:
             tmpfile.write(input_data)
         return tmpfile
+
+    def fit_afterpulse_spectrum(self):
+        """
+        Fit afterpulse spectrum with a exponential decay function.
+
+        Assume input to be in ecsv format with columns 'amplitude', 'frequency (afterpulsing)',
+        and 'frequency stdev (afterpulsing)'.
+
+        Returns
+        -------
+        astro.Table
+            Table with fitted afterpulse spectrum data.
+        """
+        ap_min = self.args_dict["afterpulse_amplitude_range"][0]
+        fix_k = self.args_dict.get("fit_afterpulse_decay_factor")
+
+        x, y, y_err = self._read_afterpulse_spectrum_for_fit(
+            self.args_dict.get("afterpulse_spectrum"), ap_min
+        )
+        fit_func, p0, bounds = self.afterpulse_fit_function(fix_k=fix_k)
+
+        result = curve_fit(fit_func, x, y, sigma=y_err, p0=p0, bounds=bounds, absolute_sigma=True)
```

Comment:
There is an assumption here that we always get uncertainties (and that they are in absolute values). Not sure we can always trust to have them, so perhaps a check would be useful.

Reply:
Is this unreasonable to require?

`_read_afterpulse_spectrum_for_fit` will fail with column "frequency stdev (afterpulsing)" not found. My assumption was that this is enough information for the user. Should I add more?

Reply:
No, that's fine. Whether it's unreasonable to require I don't know, we don't have any procedure set up for these measurements.